toolchain/check/import_cpp.cpp

```diff
@@ -18,20 +18,57 @@
 #include "toolchain/diagnostics/format_providers.h"
 
 namespace Carbon::Check {
+namespace {
+
+auto GenerateCppIncludesHeaderFilename(llvm::StringRef importing_file_path)
+    -> std::string {
+  return llvm::Twine(importing_file_path)
+      .concat(".generated.cpp_imports.h")
+      .str();
+}
+
+auto GenerateCppIncludesHeaderCode(
+    llvm::ArrayRef<std::pair<llvm::StringRef, SemIRLoc>> imports)
+    -> std::string {
+  constexpr llvm::StringLiteral IncludePrefix(R"(#include ")");
+  constexpr llvm::StringLiteral IncludeSuffix(R"("
+)");
+
+  std::string code;
+  for (const auto& [path, _] : imports) {
+    code.append(
+        llvm::Twine(IncludePrefix).concat(path).concat(IncludeSuffix).str());
+  }
+  return code;
+}
+
+}  // namespace
+
+auto ImportCppFiles(
+    Context& context, llvm::StringRef importing_file_path,
+    llvm::ArrayRef<std::pair<llvm::StringRef, SemIRLoc>> imports,
+    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs) -> void {
+  size_t num_imports = imports.size();
+  if (num_imports == 0) {
+    return;
+  }
+
+  // TODO: Use all import locations by referring each Clang diagnostic to the
+  // relevant import.
+  SemIRLoc loc = imports.back().second;
 
-auto ImportCppFile(Context& context, SemIRLoc loc,
-                   llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
-                   llvm::StringRef file_path, llvm::StringRef code) -> void {
   std::string diagnostics_str;
   llvm::raw_string_ostream diagnostics_stream(diagnostics_str);
 
   llvm::IntrusiveRefCntPtr<clang::DiagnosticOptions> diagnostic_options(
       new clang::DiagnosticOptions());
   clang::TextDiagnosticPrinter diagnostics_consumer(diagnostics_stream,
                                                     diagnostic_options.get());
+
   // TODO: Share compilation flags with ClangRunner.
   auto ast = clang::tooling::buildASTFromCodeWithArgs(
-      code, {}, file_path, "clang-tool",
+      GenerateCppIncludesHeaderCode(imports), {},
+      GenerateCppIncludesHeaderFilename(importing_file_path), "clang-tool",
```

Comment:
Looking at the helper function, it seems like the equivalent here would be:

```suggestion
      (importing_file_path + ".generated.cpp_imports.h").str(), "clang-tool",
```

Since it's a pretty short amount of code, what do you think of removing the function? You could also make a local if you prefer, like:

```
  std::string gen_header_name = (importing_file_path + ".generated.cpp_imports.h").str();
```

```suggestion
      gen_header_name, "clang-tool",
```

Reply:
Done.